### PR TITLE
fix bugs around property loading, naming, and temp graph replacement

### PR DESCRIPTION
**Pull request by jhsinger-klotho**
fix bugs around property loading, naming, and temp graph replacement

### Standard checks

- **Unit tests**: Any special considerations?
- **Docs**: Do we need to update any docs, internal or public?
- **Backwards compatibility**: Will this break existing apps? If so, what would be the extra work required to keep them working?

### CODE REVIEW

File: pkg/engine2/operational_eval/evaluator.go [hunk 1]
```diff
@@ -452,12 +453,17 @@ func (eval *Evaluator) UpdateId(oldId, newId construct.ResourceId) error {
...
-						errs = errors.Join(errs, err)
-						continue
-					}
+			}
+			if vertex.TempGraph != nil {
+				_, err := vertex.TempGraph.Vertex(oldId)
+				switch {
+				case errors.Is(err, graph.ErrVertexNotFound):
+					// do nothing
+					break
```
Comment (gordon-klotho) [inline pkg/engine2/operational_eval/evaluator.go:462]: `break`? that'd break out of the `for _, key := range topo` loop, is that what you intended?

File: pkg/engine2/testdata/k8s_api.dataflow-viz.yaml [hunk 2]
```diff
@@ -4,11 +4,11 @@ resources:
     parent: eks_cluster/eks_cluster-0
 
 
-  load_balancer/rest-api-4-integ6897f0b9:
+  load_balancer/rest-api-4-integbcc77100:
```
Comment (gordon-klotho) [inline pkg/engine2/testdata/k8s_api.dataflow-viz.yaml:7]: Is this non-deterministic?

File: pkg/engine2/operational_eval/evaluator.go (hunk 1 above)
Comment (jhsinger-klotho) [reply pkg/engine2/operational_eval/evaluator.go:462]: nope definitely didnt mean for that to be there, removing

File: pkg/engine2/testdata/k8s_api.dataflow-viz.yaml (hunk 2 above)
Comment (jhsinger-klotho) [reply pkg/engine2/testdata/k8s_api.dataflow-viz.yaml:7]: i dont think so, the ordering did change, which i assumed was from the way we load the graph, but i ran the same architecture 10 times and always saw the same result

Comment (gordon-klotho) [reply pkg/engine2/testdata/k8s_api.dataflow-viz.yaml:7]: Just wondering if the UUIDs weren't deterministic, do we set the seed somewhere?

**Review by gordon-klotho — APPROVED** [review]

File: pkg/engine2/testdata/k8s_api.dataflow-viz.yaml (hunk 2 above)
Comment (jhsinger-klotho) [reply pkg/engine2/testdata/k8s_api.dataflow-viz.yaml:7]: dont think we do, we could probably look into that

**Pull request merged**